### PR TITLE
Fix: make hyperlinks work as expected

### PR DESCRIPTION
**Pull request by RobertDHanna**
Fixes CM-556
Fixes CM-621

Test Plan:
 - Preview a course that contains content that links to other content in the description.
 - Verify that links point to what they are supposed to or point to warning pages that explain
    the preview is not available for that resource.

### CODE REVIEW

**Comment by RobertDHanna** [comment]
I made some changes, but I'd like to give it one more pass before it is ready for re-review.

**Comment by mauriciozaffari** [comment]
@RobertDHanna can we review it?

**Comment by RobertDHanna** [comment]
@mauriciozaffari I’ll have it ready to review tomorrow! Thanks for checking in!

**Comment by RobertDHanna** [comment]
@cguanzon @mauriciozaffari this commit is ready for re-review!

**Comment by RobertDHanna** [comment]
Actually, I would like to change the tests to use the already extracted cartridge. I will post here when I get that done!

File: tests/test.single-assignment-with-internal-links.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import { Selector } from "testcafe";
-
-fixture`Single assignment cartridge (with attachment)`
-  .page`http://localhost:5000/?src=https%3A%2F%2Fs3.amazonaws.com%2Fpublic-imscc%2FAssignment%2Bwith%2BInternal%2Band%2BExternal%2BLinks.imscc#/`;
+const courseUrl = `http://localhost:5000/?manifest=https://s3.amazonaws.com/public-imscc/UNZIPPED+STUFF/single-assignment-with-internal-and-external-links/imsmanifest.xml`;
+fixture`Single assignment cartridge (with attachment)`.page`${courseUrl}`;
 
 // links here will not work because of CM-620
```
[the remark this answers is not in the document]
Comment (RobertDHanna) [reply tests/test.single-assignment-with-internal-links.js]: Pushed a new change that removes this comment 👍

**Comment by RobertDHanna** [comment]
@mauriciozaffari spoke with Aaron - I will refactor the tests in a subsequent commit. It is ready for review now 👍

**Pull request merged**